packages/playwright-core/src/cli/docker.ts

```diff
@@ -0,0 +1,380 @@
+  ]);
+}
+
+export async function ensureDockerEngineIsRunningOrDie() {
+  try {
+    await callDockerAPI('get', '/info');
+  } catch (e) {
+    console.error(utils.wrapInASCIIBox([
+      `Docker is not running!`,
+      `Please install and launch docker:`,
+      ``,
+      `    https://docs.docker.com/get-docker`,
+      ``,
+    ].join('\n'), 1));
+    process.exit(1);
+  }
+}
+
+async function findDockerImage(imageName: string): Promise<DockerImage|undefined> {
+  const images: DockerImage[] | null = await getJSON('/images/json');
+  return images ? images.find(image => image.RepoTags?.includes(imageName)) : undefined;
+}
+
+interface Container {
+  ImageID: string;
+  State: string;
+  Names: [string];
+  Id: string;
+}
+
+async function findRunningDockerContainerId(): Promise<string|undefined> {
+  const containers: (Container[]|undefined) = await getJSON('/containers/json');
+  if (!containers)
+    return undefined;
+  const dockerImage = await findDockerImage(VRT_IMAGE_NAME);
+  const container = dockerImage ? containers.find((container: Container) => container.ImageID === dockerImage.Id) : undefined;
+  return container?.State === 'running' ? container.Id : undefined;
+}
+
+interface ContainerOptions {
+  image: DockerImage;
+  autoRemove: boolean;
+  command?: string[];
+  ports?: Number[];
+  name?: string;
+}
+
+async function launchContainer(options: ContainerOptions): Promise<string> {
+  const ExposedPorts: any = {};
+  const PortBindings: any = {};
+  for (const port of (options.ports ?? [])) {
+    ExposedPorts[`${port}/tcp`] = {};
+    PortBindings[`${port}/tcp`] = [{ HostPort: port + '' }];
+  }
+  const container = await postJSON(`/containers/create` + (options.name ? '?name=' + options.name : ''), {
+    Cmd: options.command,
+    AttachStdout: true,
+    AttachStderr: true,
+    Image: options.image.Id,
+    ExposedPorts,
+    HostConfig: {
+      Init: true,
+      AutoRemove: options.autoRemove,
+      ShmSize: 2 * 1024 * 1024 * 1024,
+      PortBindings,
+    },
+  });
+  await postJSON(`/containers/${container.Id}/start`);
+  return container.Id;
+}
+
+async function getJSON(url: string): Promise<any> {
+  const result = await callDockerAPI('get', url);
+  if (!result)
+    return result;
+  return JSON.parse(result);
+}
+
+async function postJSON(url: string, json: any = undefined) {
+  const result = await callDockerAPI('post', url, json ? JSON.stringify(json) : undefined);
+  if (!result)
+    return result;
+  return JSON.parse(result);
+}
+
+const DOCKER_API_VERSION = '1.41';
+
+function callDockerAPI(method: 'post'|'get'|'delete', url: string, body: Buffer|string|undefined = undefined): Promise<string> {
+  const dockerSocket = process.platform === 'win32' ? '\\\\.\\pipe\\docker_engine' : '/var/run/docker.sock';
+  return new Promise((resolve, reject) => {
+    const request = http.request({
+      socketPath: dockerSocket,
+      path: `/v${DOCKER_API_VERSION}${url}`,
+      method,
```

Comment:
```suggestion
      timeout: NET_DEFAULT_TIMEOUT
      method,
```
import it from this file: https://github.com/mxschmitt/playwright/blob/fix-doclint/packages/playwright-core/src/common/netUtils.ts#L42

Reply:
Let me hard code 30 seconds timeout here so that I don't need to export more things from `playwright-core` (the `docker.ts` file got moved into playwright-test)